@xen-orchestra/xapi/src/vm.js

```diff
@@ -426,6 +427,7 @@ module.exports = class Vm {
       query.sr_id = srRef
     }
     if (onVmCreation != null) {
+      onVmCreation = once(onVmCreation)
```

Comment:
Not sure you need to use `once`, it would be interesting to remember if `onVmCreation` has been called or not so that we avoid unnecessarily calling `this.getRecords('VM', ref)`.

Also, we probably need to remove the watcher if we know it will never gonna be called.

Reply:
Currently, there is no way to remove a generic watcher.

Reply:
Let's add it then.